src/WL/ServerCore.hsc

```diff
@@ -8,6 +8,11 @@ import Foreign.C.String
 import WL.Utils
 import WL.ServerProtocol
 
+-- Opaque objects
+type WL_event_source = ()
+type WL_global = ()
+type WL_resource = ()
+
 type WL_notify_func_t
```

Comment:
I defined these as proper `data` definitions, defined to be equal to `struct wl_global` etc.